roles/cloud_providers/tasks/aws/ec2_instances.yml

```diff
@@ -0,0 +1,115 @@
+- name: find ami instance-id for RHEL7
+  ec2_ami_info:
+    owners: "aws-marketplace"
+    filters:
+       "product-code": "aw0evgkw8e5c1q413zgy5pjce"
```

Comment:
can we move this also the group vars?. Also can we pull the image using name?

Reply:
also is the product code same across regions?. If not we should stick to `name`

Reply:
As far as I am concerned, this product code is available across all regions since it is available on aws-marketplace and is not a region specific AMI (the **same product code is in USA** regions for example in comparison with an AMI based search that is frequently attached to a region). The [documentation](https://wiki.centos.org/Cloud/AWS) clarifies which product codes can be used for "CentOS 7 (x86_64) - with Updates HVM". Also, I made sure to add an automation process to extract the latest version from the image based on the product code.

Reply:
Regarding, let me add a variable within **inventories/inventory_aws/group_vars** folder to refer the CentOS product code (which is a unique identifier as I have noticed from the AWS market place).

Reply:
More information on AWS Marketplace:
[AMI-products](https://docs.aws.amazon.com/marketplace/latest/userguide/ami-products.html)

Reply:
I added variable for the AWS Marketplace product code on the group_vars for the inventory cloud specific folder:
`instance_product_code: "aw0evgkw8e5c1q413zgy5pjce"`